### PR TITLE
Small HttpRequestMessageFactory improvements

### PR DESCRIPTION
**Pull request by danielmarbach**
This PR improves slightly the HttpRequestMessageFactory

## Description

- Remove an unnecessary async call
- Switches to the contains method on the hashset instead of using LINQ
- Uses `TryGetValue` instead of `ContainsKey` followed by an indexer access

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs [hunk 1]
```diff
@@ -474,14 +474,7 @@ public void SetRequestHeaders(IDictionary<string, string> headers)
         /// <returns>The request content.</returns>
         public HttpContent GetRequestContent()
```
Comment (normj) [inline sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs:475]: Turning this into a simple `return` is triggering the build error [CA1024](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1024). I don't want to change this public method so can you wrap the method with `#pragma warning disable CA1024` and `#pragma warning restore CA1024`?

**Review by normj — CHANGES_REQUESTED** [review]

File: sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs (hunk 1 above)
Comment (danielmarbach) [reply sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs:476]: Done

**Review by normj — APPROVED** [review]
Looks good and I did a full dry run build that was successful.

**Review by dscpinheiro — APPROVED** [review]

**Comment by danielmarbach** [comment]
Do you need anything else from my side to get this merged?

**Comment by normj** [comment]
@danielmarbach No, it is queued up to go out in the next non-service update release.

**Pull request closed without merging**